src/MongoDB/Session.c

```diff
@@ -0,0 +1,275 @@
+	PHP_ME(Session, getClusterTime, ai_Session_void, ZEND_ACC_PUBLIC|ZEND_ACC_FINAL)
+	PHP_ME(Session, getLogicalSessionId, ai_Session_void, ZEND_ACC_PUBLIC|ZEND_ACC_FINAL)
+	ZEND_NAMED_ME(__construct, PHP_FN(MongoDB_disabled___construct), ai_Session_void, ZEND_ACC_PRIVATE|ZEND_ACC_FINAL)
+	ZEND_NAMED_ME(__wakeup, PHP_FN(MongoDB_disabled___wakeup), ai_Session_void, ZEND_ACC_PUBLIC|ZEND_ACC_FINAL)
+	PHP_FE_END
+};
+/* }}} */
+
+/* {{{ MongoDB\Driver\Session object handlers */
+static zend_object_handlers php_phongo_handler_session;
+
+static void php_phongo_session_free_object(phongo_free_object_arg *object TSRMLS_DC) /* {{{ */
+{
+	php_phongo_session_t *intern = Z_OBJ_SESSION(object);
+
+	zend_object_std_dtor(&intern->std TSRMLS_CC);
+
+	if (intern->client_session) {
+		mongoc_client_session_destroy(intern->client_session);
+	}
+
+#if PHP_VERSION_ID < 70000
+	efree(intern);
+#endif
+} /* }}} */
+
+static phongo_create_object_retval php_phongo_session_create_object(zend_class_entry *class_type TSRMLS_DC) /* {{{ */
+{
+	php_phongo_session_t *intern = NULL;
+
+	intern = PHONGO_ALLOC_OBJECT_T(php_phongo_session_t, class_type);
+
+	zend_object_std_init(&intern->std, class_type TSRMLS_CC);
+	object_properties_init(&intern->std, class_type);
+
+#if PHP_VERSION_ID >= 70000
+	intern->std.handlers = &php_phongo_handler_session;
+
+	return &intern->std;
+#else
+	{
+		zend_object_value retval;
+		retval.handle = zend_objects_store_put(intern, (zend_objects_store_dtor_t) zend_objects_destroy_object, php_phongo_session_free_object, NULL TSRMLS_CC);
+		retval.handlers = &php_phongo_handler_session;
+
+		return retval;
+	}
+#endif
+} /* }}} */
+
+static HashTable *php_phongo_session_get_debug_info(zval *object, int *is_temp TSRMLS_DC) /* {{{ */
+{
+	php_phongo_session_t       *intern = NULL;
+	const mongoc_session_opt_t *cs_opts;
+#if PHP_VERSION_ID >= 70000
+	zval                        retval;
+#else
+	zval                        retval = zval_used_for_init;
+#endif
+
+	*is_temp = 1;
+	intern = Z_SESSION_OBJ_P(object);
+
+	array_init(&retval);
+
+	{
+		const bson_t *lsid;
+
+		php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+		/* Use native arrays for debugging output */
+		state.map.root_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
+		state.map.document_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
+
+		lsid = mongoc_client_session_get_lsid(intern->client_session);
+
+		php_phongo_bson_to_zval_ex(bson_get_data(lsid), lsid->len, &state);
+
+#if PHP_VERSION_ID >= 70000
+		ADD_ASSOC_ZVAL_EX(&retval, "logicalSessionId", &state.zchild);
+#else
+		ADD_ASSOC_ZVAL_EX(&retval, "logicalSessionId", state.zchild);
+#endif
+	}
+
+	{
+		const bson_t *cluster_time;
+
+		php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+		/* Use native arrays for debugging output */
+		state.map.root_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
+		state.map.document_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
```

Comment:
You'd almost want to pull this often done conversion with NATIVE/NATIVE out into its own function.

Reply:
Or perhaps a separate initializer struct like `PHONGO_BSON_STATE_INITIALIZER` but used just for debugging? Either way, that's a good idea. I've opened [PHPC-1074](https://jira.mongodb.org/browse/PHPC-1074) to track it.